skimage/transform/_distance_transform.pyx

```diff
@@ -0,0 +1,143 @@
+import numpy as np
+import warnings
+cimport numpy as np
+
+cdef f(bint p):
+    cdef double out
+    if p == 0:
+        out = 0
+    out = np.finfo(np.float64).max
+    print(warnings.warn(p,out))
+    return out
+
+cdef euclidean_dist(double a, double b, double c):
```

Comment:
These functions should be declared `nogil`, and the gil should be be released before calling them.

```
 cdef euclidean_dist(double a, double b, double c) nogil: 
```

If you enable maintainers to edit your branch, we can use some more of git's suggestion features.